packages/ens-utils/src/registrar.ts

```diff
@@ -0,0 +1,356 @@
+import { buildAddress } from "./address";
+import { MAINNET } from "./chain";
+import { buildContractRef, ContractRef } from "./contract";
+import { ENSName } from "./ensname";
+import { NFTIssuer } from "./nft";
+import { approxScalePrice, Price } from "./price";
+import { Duration, SECONDS_PER_YEAR, TimePeriod, Timestamp } from "./time";
+
+// REGISTRATION STATUSES ⬇️
+
+// TODO: make more generic to support non .eth 2nd level domains
+export enum PrimaryRegistrationStatus {
+  Active = "Active",
+  Expired = "Expired",
+  NeverRegistered = "NeverRegistered",
+}
+
+// TODO: make more generic to support non .eth 2nd level domains
+export enum SecondaryRegistrationStatus {
+  ExpiringSoon = "ExpiringSoon",
+  FullyReleased = "FullyReleased",
+  GracePeriod = "GracePeriod",
+  RecentlyReleased = "RecentlyReleased",
+}
+
+// TODO: make more generic to support non .eth 2nd level domains
+export type Registration = {
+  // Below timestamps are counted in seconds
+  registrationTimestamp: Timestamp | null;
+  expirationTimestamp: Timestamp | null;
+  expiryTimestamp: Timestamp | null; // TODO: Franco, could you please remove this for us?
```

Comment:
Yes, I will handle it [here](https://app.shortcut.com/ps-web3/story/25399/remove-duplicate-expirytimestamp-field)

```suggestion
  expiryTimestamp: Timestamp | null;
```